### DISCUSSION TITLE
Native View Embedding with Tauri

### DISCUSSION THREAD

**Discussion by velocitysystems** [tauri-apps/tauri#11918] [General]
Hello! 👋 I’m currently evaluating Tauri to replace React Native for a cross platform iOS, Android and Windows app.

With React Native it is possible to _integrate_ a native view component inside the rendered view e.g. Video player control which is natively implemented on all three platforms. Is this possible in Tauri with Tao? The business case is a highly-customized video player which is written in Swift, Kotlin and C# (UWP).

From my research I understand the answer to this question to be "no". Is that correct? Are there any alternative approaches where native views can be rendered inside a webview with Tao/Wry?

**Comment by FabianLars** [comment]
It's kindaaa possible, but so far only on desktop (may require the `unstable` feature flag). For example, you can have multiple webviews in one window, or render directly inside of a window with the [wgpu crate](https://github.com/clearlysid/tauri-wgpu-cam). I'm pretty sure i've also seen some mpv and/or gstreamer embedds here or on discord (though can't find it rn).

Think of them as layers more so than elements/views.

While multi webview support is planned for mobile, i think we've not thought about some native-ish embedds.

**Comment by velocitysystems** [comment]
Thank you for the helpful resources @FabianLars.
